types/map4dsdk/index.d.ts

```diff
@@ -0,0 +1,187 @@
+// Definitions by: Sua Le <https://github.com/sua8051>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module 'map4d' {
+    interface CircleOptions {
+        center: LatLngLike
+        radius?: number
+        fillColor?: string
+        fillOpacity?: number
+        visible?: boolean
+    }
+
+    class Circle {
+        constructor(options: CircleOptions)
+        getMap(): Map
+        setMap(map: Map): void
+    }
+
+    class Icon {
+        constructor(width: number, height: number, url: string)
+        clone(): Icon
+        getWidth(): number
+        getHeight(): number
+        getUrl(): string
+    }
+
+    type LatLngLike = LatLng | { lat: number, lng: number } | [number, number]
+    class LatLng {
+        constructor(lat: number, lng: number)
+        equals(other: LatLng): boolean
+        lat(): number
+        lng(): number
+        toString(): string
+    }
+
+    type LatLngBoundsLike = LatLngBounds | [LatLngLike, LatLngLike] | [number, number, number, number]
+    class LatLngBounds {
+        constructor(sw: LatLngLike, ne: LatLngLike)
+        getCenter(): LatLng
+        getNortheast(): LatLng
+        getSouthwest(): LatLng
+    }
+
+    enum ControlOptions {
+        TOP_LEFT,
+        TOP_RIGHT,
+        BOTTOM_LEFT,
+        BOTTOM_RIGHT
+    }
+
+    interface MapOptions {
+        center?: [number, number]
+        zoom?: number
+        minZoom?: number
+        maxZoom?: number
+        geolocate?: boolean
+        accessKey?: string
+        tilt?: number
+        mapControls?: boolean
+        mapControlOptions?: ControlOptions
+    }
+
+    interface AnimationOptions {
+        duration?: number
+        easing?: (arg0: number) => number
+        animate?: boolean
+    }
+
+    interface PaddingOptions {
+        top?: number
+        bottom?: number
+        left?: number
+        right?: number
+    }
+
+    class Map {
+        constructor(container: HTMLElement, options?: MapOptions)
+        enable3dMode(enabled: boolean): void
+        getCenter(): LatLng
+        setCenter(latLng: LatLng): void
+        setOptions(options: MapOptions): void
+        getZoom(): number
+        setZoom(zoom: number): void
+        getTilt(): number
+        setTilt(tilt: number): void
+        setMapUrl(mapUrl: string): void
+        getMapUrl(): string
+        setMinZoom(minZoom: number): void
+        setMaxZoom(maxZoom: number): void
+        panBy(offset: PointLike, animationOptions?: AnimationOptions): void
+        panTo(latLng: LatLngLike, animationOptions?: AnimationOptions): void
+        flyTo(latLng: LatLngLike, zoom: number, animationOptions?: AnimationOptions): void
+        fitBounds(bounds: LatLngBoundsLike, padding?: PaddingOptions, animationOptions?: AnimationOptions): void
+        addListener(eventName: string, func: Function): MapsEventListener
```

Comment:
Please don't use the type `Function`, as discussed [here](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes)